swarm/src/behaviour.rs

```diff
@@ -152,6 +152,13 @@ type THandlerInEvent<THandler> =
 /// struct members are delegated to [`NetworkBehaviourEventProcess`] implementations. Those must be
 /// provided by the user on the type that [`NetworkBehaviour`] is derived on.
 ///
+/// Because these implementations are granted exclusive access to the [`NetworkBehaviour`],
+/// [blocking code](https://ryhl.io/blog/async-what-is-blocking/) in these implementations will
+/// block the entire [`Swarm`](crate::Swarm) from processing new events, since the swarm cannot progress
+/// without also having exclusive access to the [`NetworkBehaviour`]. A better alternative is to spawn
+/// blocking or asynchronous tasks on a (separate) executor, which may be provided to the behaviour's
+/// constructor.
```

Comment:
```suggestion
/// without also having exclusive access to the [`NetworkBehaviour`]. A better alternative is to execute
/// blocking or asynchronous logic on a separate task.
```

I don't think spawning a new task on events is the right approach for most applications. You would need some external back-pressure mechanism, in other words you need some mechanism to stop you from spawning a large amount of tasks. (Yes they are cheap, still they break back-pressure.)

How about the wording above? If you want to execute your blocking logic somewhere else, I would suggest having a dedicated task or a limited set of dedicated tasks.

Reply:
Thanks, that seems like a better idea. I added the concrete suggestion to use a bounded channel, so that it's more obvious for new users how they could accomplish such a thing.